File: search engine svc/src/db_actions.js
const DBSOURCE = "../../data/crawled_web.db";
let sqlite3 = require('sqlite3').verbose();
let db = new sqlite3.Database(DBSOURCE);

let GetData = function (queryString, retFunc) {
    let sql = "select title,url, (select count(1) from result where title like '%' || ? || '%') as count from result where title like '%' || ? || '%' limit 100";
    let params = [queryString, queryString];
    db.all(sql, params, (err, rows) => {
        if (err) {
            retFunc([{ "error": err.message }]);
        }
        retFunc(rows);
    });
}

module.exports.GetData = GetData;

File: scrapper/src/db_actions.py
import sqlite3

conn = sqlite3.connect('../../data/crawled_web.db')
_cursor = conn.cursor()

def insert_result(params):
    sql = "INSERT INTO result VALUES (?, ?, ?)"
    _cursor.execute(sql, (params))

def get_intial_url_load():
    sql = "SELECT url FROM result"
    rows = _cursor.execute(sql)
    return rows.fetchall()